### ISSUE TITLE
Covers that control the machine should have AND-logic to enable the machine

### ISSUE THREAD

**Issue by dahgul333** [GTNewHorizons/GT-New-Horizons-Modpack#11843]
### Your GTNH Discord Username

@dahgul#1726

### Your Pack Version

2.2.0.0

### Your Server

Eta

### Your Expectation

When i try aftomate another pice of my chem. I put Redstone reciver(in) cover
![image](https://user-images.githubusercontent.com/35747549/202580996-57b2405a-6965-4307-83d7-d69becde4ead.png)
on a side of LCR(controller) and it turn-on 

### The Reality

It can not will be tur-offed until cover is removed, you can't do this with hammer or with Machine controller cover
![image](https://user-images.githubusercontent.com/35747549/202581782-ad401998-e0a6-441e-a82d-bde12ee7c4c1.png)
![image](https://user-images.githubusercontent.com/35747549/202581800-24fa394b-a41b-443a-91db-92527499ccd5.png)
Sometimes for playback you need to configurate redstone cover

### Your Proposal

fix.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [X] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by minecraft7771** [comment]
This is not a bug. The machine receives a signal to work and it does. It has OR-logic, it won't try to search for all covers it has. I'll change this to a suggestion to change it to AND-logic.

**Comment by chochem** [comment]
you can just invert the signals if you want AND instead of OR. No need to change the game. OR makes a lot of sense as the default behaviour.

However, I am not sure thats really what the ticket is about. The English is a bit hard to follow...

**Comment by Godelboard** [comment]
I think what is trying to be said here is that machine controller cover and redstone receiver (in) conflict with each other. The other week I was trying to use them to do something and wanted to make sure it worked before implementing, so I built a test scenario table and I found they did not play nice together. Here are the results:

- RS Receiver was toggled via lever on a RS Transmitter on a nearby wooden frame box.
- Machine controller was set to "active with redstone" and toggled by lever.
- For both, on/off references the state of the lever.
    - X = Cover not on machine
    - 0 = Cover on machine and off
    - 1 = Cover on machine and on

|  RS Receiver | Machine Controller | Machine Behavior |
| --- | --- | --- |
|X|X|Mallet Status|
|X|0|Off|
|X|1|On|
|0|X|Off|
|0|0|Off|
|0|1|On (behavior conflict)|
|1|X|On|
|1|0|Off (behavior conflict)|
|1|1|On|

As you can see on two rows I called out, the machine completely disregards what the RS receiver is sending and only listens to the machine controller. Therefor, having a wireless RS cover and machine controller together doesn't work. I believe that's what this post is highlighting and saying. In any case, it's neither AND or OR logic being applied, it's "ignore RS receiver" logic.

Background: I wanted to have a "master switch" for all the MBs in our base that we could toggle when changing power infrastructure rather than having to have batteries everywhere or go around all over the base turning everything back on after changing LSC hatches/capacitors/etc.

**Comment by Glease** [comment]
Why this kind of conflicts again... The most likely resolution is to not allow machine controller and rs receiver cover on the same block, as we did last time when the question of interaction of multiple machine controller cover on the same machine is brought up.
---
BTW it's not "ignore RS receiver logic". It's "ignore all machine controlling cover except the last one". Which is the last one is an implementation detail that might change without notice.

**Comment by Godelboard** [comment]
That would check out as expected behavior then as I believe I had applied the machine controller last? Don't really remember.

But anyways personally I abandoned the idea of having a master switch to pause all MBs during power upgrades and will just accept the solution accepted in Discord which was "just be fast 5head" and go do a base tour when it fails. I'm not particular invested in the decision for how it works going either way.

I did see the advanced wireless covers have ability to combine and do logic on multiple signals *in the same frequency*, but frankly I don't fancy setting up and maintaining a unique ID for each MB that's going to be controlled based on the fullness of the output bus/hatch and would rather just run a few pieces of conduit instead.
